Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -41,61 +37,64 @@ uint32_t canID;
 // CAN variables
 uint8_t TxData[8] = {0};
 uint8_t RxData[8];
-uint8_t HexToDecData[8] = {0};
 uint32_t TxMailbox;
+uint32_t CanID;
+float DecodeDataFrame[8] = {0};
 bool CANMessageReceivedFlag = false;
 
+/*
+	this is the CAN ID list for the PE3 8400 which can also be viewed from the link below
+	https://pe-ltd.com/assets/AN400_CAN_Protocol_C.pdf
+	the link also tells you what information is inside each of the CAN ID
+	NOTE: the Id list is in the same order as the application note from top to bottom
+
+*/
+constexpr uint32_t CanIDList[16] = {0x0CFFF048, 0x0CFFF148, 0x0CFFF248, 0x0CFFF348,
+						  	  	  	0x0CFFF448, 0x0CFFF548, 0x0CFFF648, 0x0CFFF748,
+									0x0CFFF848, 0x0CFFF948, 0x0CFFFA48, 0x0CFFFB48,
+									0x0CFFFC48, 0x0CFFFD48, 0x0CFFFE48, 0x0CFFD048 };
+
 // CallBack functions
 
 
-//Can CallBack
+					//Can CallBack
 void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef *hcan1)
 {
 
-
     if (HAL_CAN_GetRxMessage(hcan1, CAN_RX_FIFO0, &RxHeader, RxData) == HAL_OK)
     {
     	CANMessageReceivedFlag = true;
-    	canID = RxHeader.ExtId;
+    	CanID = RxHeader.ExtId;
 
 
     }
 }
 
-void cppMain() {
+// function prototypes
 
+void canFilterSetup(int FilterBankNum, uint32_t FilterExtID );
+void canParseDataFrame();
+void canDecodeDataFrame(float DecodeDataFrame[],int DataSet );
 
-	// Filter initialization
-	CAN.FilterIdHigh = 0x0CFFF048 >> 13 ;
-	CAN.FilterIdLow = ((0x0CFFF048 & 0x1FFF) <<3) | CAN_ID_EXT | CAN_RTR_DATA ;
-	CAN.FilterMaskIdHigh = 0xFFFF;
-	CAN.FilterMaskIdLow = 0xFFFF;
-	CAN.FilterFIFOAssignment = CAN_FILTER_FIFO0;
-	CAN.FilterBank = 10;
-	CAN.FilterMode = CAN_FILTERMODE_IDMASK;
-	CAN.FilterScale = CAN_FILTERSCALE_32BIT;
-	CAN.FilterActivation = CAN_FILTER_ENABLE;
-	CAN.SlaveStartFilterBank = 13;
-	HAL_CAN_ConfigFilter(&hcan1,&CAN);
-	// Filter initialization
-	CAN_two.FilterIdHigh = 0x0CFFF548 >> 13 ;
-	CAN_two.FilterIdLow = ((0x0CFFF548 & 0x1FFF) <<3) | CAN_ID_EXT | CAN_RTR_DATA ;
-	CAN_two.FilterMaskIdHigh = 0xFFFF;
-	CAN_two.FilterMaskIdLow = 0xFFFF;
-	CAN_two.FilterFIFOAssignment = CAN_FILTER_FIFO0;
-	CAN_two.FilterBank = 9;
-	CAN_two.FilterMode = CAN_FILTERMODE_IDMASK;
-	CAN_two.FilterScale = CAN_FILTERSCALE_32BIT;
-	CAN_two.FilterActivation = CAN_FILTER_ENABLE;
-	CAN_two.SlaveStartFilterBank = 13;
-	HAL_CAN_ConfigFilter(&hcan1,&CAN_two);
-
-	// handling TX message
-	TxHeader.ExtId = 0x123;
-	TxHeader.DLC = 8;
-	TxHeader.IDE = CAN_ID_EXT;
-	TxHeader.RTR = CAN_RTR_DATA;
-	TxHeader.TransmitGlobalTime = DISABLE;
+void cppMain() {
+
+	// initializing all filters
+	canFilterSetup(1,CanIDList[0]);
+	canFilterSetup(2,CanIDList[1]);
+	canFilterSetup(3,CanIDList[2]);
+	canFilterSetup(4,CanIDList[3]);
+	canFilterSetup(5,CanIDList[4]);
+	canFilterSetup(6,CanIDList[5]);
+	canFilterSetup(7,CanIDList[6]);
+	canFilterSetup(8,CanIDList[7]);
+	canFilterSetup(9,CanIDList[8]);
+	canFilterSetup(10,CanIDList[9]);
+	canFilterSetup(11,CanIDList[10]);
+	canFilterSetup(12,CanIDList[11]);
```

Comment:
I configured all the filters from the ECU for demonstration purposes. Obviously, we don't need all the filters here but I just wanted to show that the code could handle all the filters that  would be needed if we wanted to push the ECU's CAN bus to its limits